lib/grant.js

```diff
@@ -285,14 +295,16 @@ function useClientCredentialsGrant (done) {
   }
 
   var self = this;
-  return this.model.getUserFromClient(clientId, clientSecret,
-      function (err, user) {
+  return this.model.getUserFromClient(clientId, clientSecret, function (err, user) {
     if (err) return done(error('server_error', false, err));
+
     if (!user) {
       return done(error('invalid_grant', 'Client credentials are invalid'));
     }
 
     self.user = user;
+    self.scope = self.client.scope;
```

Comment:
I think this should be done on a per request basis and so should be read from the request?